src/persist/src/metrics.rs

```diff
@@ -139,4 +139,11 @@ impl ColumnarMetrics {
 
         Self::new(&registry, &lgbytes, cfg, false)
     }
+
+    /// Returns a reference to the inner [`ConfigSet`].
```

Comment:
Oof! It's already a pretty big hack to have this ConfigSet here, exposing it publicly feels over a line to me.

Two alternatives:
- It should work just as well to update the `ConfigSet` in `PersistConfig` that we passed in to `ColumnarMetrics::new` via `Metrics::new`
- The `ConfigSet` toggling _should_ work in the cli tool impl because it's all single threaded, but feels a bit brittle. We could make `decode_trace_parquet` into a thin wrapper around getting the dyncfg value and calling either `decode_trace_parquet_arrow2` or `decode_trace_parquet_arrow_rs` and use those directly in the cli tool (and ditto for encode)

Reply:
Ah ha, that's nice!

Do you think we should update [this comment](https://github.com/MaterializeInc/materialize/blob/main/src/dyncfg/src/lib.rs#L188-L189)? It led me to believe that since `ColumnarMetrics` held a cloned `ConfigSet` its values would get updated independently of the `ConfigSet` it was cloned from

Reply:
Ah, yeah, I can see the confusion! A `ConfigSet` that is cloned shares state with the original, but a new `ConfigSet` is totally independent of any others